types/node/index.d.ts

```diff
@@ -797,15 +804,18 @@ declare module "http" {
     /**
      * @deprecated Use IncomingMessage
      */
-    export interface ServerRequest extends IncomingMessage {
+    export class ServerRequest extends IncomingMessage {
         connection: net.Socket;
     }
 
     export interface ServerResponseHeaders {
       [key: string]: number | string | string[];
     }
 
-    export interface ServerResponse extends stream.Writable {
+    // https://github.com/nodejs/node/blob/master/lib/_http_server.js#L108
+    export class ServerResponse extends stream.Writable {
```

Comment:
`ServerResponse` inherits from `OutgoingMessage` (see https://github.com/nodejs/node/blob/master/lib/_http_server.js#L122).
I think a new export `OutgoingMessage` should be added here and used as base for `ServerResponse` and `ClientRequest`. see also #17720